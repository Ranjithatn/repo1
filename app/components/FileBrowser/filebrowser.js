import React, { Fragment } from "react";
import Icon from "../Icon/Icon";
import "./FileBrowser.scss";

const FileBrowser = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  onChange,
  ref,
  className,
  fileStyle,
}) => {
  return (
    <div className="fileUpload button is-primary is-medium">
      {leftIcon && <Icon icon={leftIcon} />}
      <span>{text}</span>
      {rightIcon && <Icon icon={rightIcon} />}
      <input id="myInput"
        type="file"
        className={"upload" + (className !== undefined ? " " + className : "")}
        ref={ref}
        onChange={onChange}
        onClick={onClick}
        style={fileStyle || {}}
      />
    </div>
  );
};

export default FileBrowser;
