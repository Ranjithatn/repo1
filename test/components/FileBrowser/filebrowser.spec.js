import React from "react";
import toJson from "enzyme-to-json";
import FileBrowser from "../../../app/components/FileBrowser/filebrowser";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<Figure />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <FileBrowser />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
