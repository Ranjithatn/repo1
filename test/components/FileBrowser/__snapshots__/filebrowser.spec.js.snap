// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Figure /> should render without crashing 1`] = `
<FileBrowser>
  <div
    className="fileUpload button is-primary is-medium"
  >
    <span />
    <input
      className="upload"
      id="myInput"
      style={Object {}}
      type="file"
    />
  </div>
</FileBrowser>
`;
